castle/cms/browser/security/login.py

```diff
@@ -295,9 +296,31 @@ def send_authorization(self):
             'state': new_state
         })
 
+    def check_whitelist(self):
+        username = self.username
+        if '@' in username:
```

Comment:
I suggest switching this to use [email.utils.parseaddr](https://docs.python.org/2/library/email.utils.html) instead of just splitting on "@". This will likely help take care of some of the corner cases to an email address and simplify the code a little.